users/views.py

```diff
@@ -18,11 +19,16 @@ def user_profile(request):
             sections_list = Section.objects.filter(teacher=user.id)
             return render(request, 'users/profile.html', {'user': user, 'sections_list': sections_list})
         elif user.is_student():
+            due_assignments_list = []
             sections_list = Section.objects.filter(students=user.id)
             for section in sections_list:
                 assignments_list.extend(Assignment.objects.filter(section=section))
+            assignments_list.sort()
+            for assignment in assignments_list:
+                if assignment.due_date > timezone.now():
+                    due_assignments_list.append(assignment)
```

Comment:
might want to add a function `due()` to the assignment model, but that is just cleanup. I don't think we can make this faster.